source/framework/core/src/TRestHits.cxx

```diff
@@ -546,28 +546,165 @@ Double_t TRestHits::GetSigmaY() {
 
     return sigmaY = TMath::Sqrt(sigmaY2);
 }
-
-Double_t TRestHits::GetGaussSigmaX(Int_t readoutChannels, Int_t startChannel, Int_t endChannel, Double_t pitch) {
+/**
+Double_t TRestHits::GetGaussSigmaX(Int_t nBins = 100) {
 	TH1::AddDirectory(kFALSE);
     Double_t gausSigmaX = 0;
 
-	TH1D* hX = new TH1D("hitsGaussHistoX", "hitsGaussHistoX", readoutChannels, startChannel, endChannel);
+     //double min = *min_element(fX.begin(), fX.end());
+     //double max = *max_element(fX.begin(), fX.end());
+
 
+   /*
+	TH1D* hX = new TH1D("hitsGaussHistoX", "hitsGaussHistoX", readoutChannels, startChannel, endChannel);
+	//hX->Sumw2();
     for (int n = 0; n < GetNumberOfHits(); n++) hX->Fill(fX[n], fEnergy[n]);
+	*//**
+	for (int n = 0; n < GetNumberOfHits(); n++) {
+		hX->Fill(fX[n], fEnergy[n]);
+		cout << "fX[n] is " << fX[n] << endl;
+		cout << "energy is: " << fEnergy[n] << endl;
+	}
+	**/
+	//WriteHitToTextFile("/home/cristina/hits");
+   	/*TCanvas *c = new TCanvas("c","X position fit",200,10,500,500);
+	hX->Draw();
 
     TF1* fit = new TF1("", "gaus", hX->GetMaximumBin() / (1/pitch) - 32, hX->GetMaximumBin() / (1/pitch) - 26);
 
-    hX->Fit(fit, "QNRL");  // Q = quiet, no info in screen; N = no plot; R = fit in the function range; L = log likelihood fit
+    hX->Fit(fit, "QRL");  // Q = quiet, no info in screen; N = no plot; R = fit in the function range; L = log likelihood fit (default is chi-square) ; W = Set all weights to 1 for non empty bins; ignore error bars; WW = Set all weights to 1 including empty bins; ignore error bars
+	
+	//cout << "Drawing fit" << endl;	
+	//fit->Draw();
+
+	c->Update();
 
     gausSigmaX = fit->GetParameter(2);
 
-    delete fit;
-	delete hX;
+    //delete fit;
+	//delete hX;
 
 	TH1::AddDirectory(kTRUE);
+    
     return gausSigmaX;
-}
-
+     */
+    //return max;
+//}
+/**
+void TRestHits::WriteHitToTextFile(TString filename) {
+    FILE* fff = fopen(filename.Data(), "w");
+    for (int i = 0; i < GetNumberOfHits(); i++)
+        fprintf(fff, "%d\t%e\t%e\n", i, fX[i], fEnergy[i]);
+    fclose(fff);
+}
+**/
+
+Double_t TRestHits::GetGaussSigmaX() {
+	Double_t gausSigmaX = 0;
+	Int_t nHits = GetNumberOfHits();
+	//cout << "nHits = " << n << endl;
+	Double_t x[nHits], y[nHits], ex[nHits], ey[nHits];
+	if (nHits <= 3) {	
+		gausSigmaX = 0;
+	} else {
+	//Data to fill the graph
+		for (int n = 0; n < GetNumberOfHits(); n++) {
+			//if ((fType.size() == 0 ? !IsNaN(fX[n]) : fType[n] % X == 0)){
+				//hX->Fill(fX[n], fEnergy[n]);
+				x[n] = fX[n];
+				y[n] = fEnergy[n];
+				//cout << "fX[n] = " << fX[n] << endl;
+				//cout << "energy = " << fEnergy[n] << endl;
+				//Including the errors
+				ex[n] = 0;
+				if (y[n] != 0) {
+					ey[n] = 11.2*sqrt(y[n]);		
+				} else {
+					ey[n] = 0; 
+				}
+			//}	
+		}	
+		//Creating TGraph
+		TGraphErrors *grX = new TGraphErrors(nHits,x,y,ex,ey);
+		//TCanvas *c = new TCanvas; //DEBUG
```

Comment:
Try to reduce the commented lines. Keep just the ones you really think will help others, and yourself in a future to read and/or understand the code.